node/src/accountant/scanners/mod.rs

```diff
@@ -96,25 +108,280 @@ impl Scanners {
+    }
+
+    pub fn start_receivable_scan_guarded(
+        &mut self,
+        wallet: &Wallet,
+        timestamp: SystemTime,
+        response_skeleton_opt: Option<ResponseSkeleton>,
+        logger: &Logger,
+        automatic_scans_enabled: bool,
+    ) -> Result<RetrieveTransactions, StartScanError> {
+        let triggered_manually = response_skeleton_opt.is_some();
+
+        if triggered_manually && automatic_scans_enabled {
+            return Err(StartScanError::ManualTriggerError(
+                MTError::AutomaticScanConflict,
+            ));
+        }
+        if let Some(started_at) = self.receivable.scan_started_at() {
+            return Err(StartScanError::ScanAlreadyRunning {
+                pertinent_scanner: ScanType::Receivables,
+                started_at,
+            });
+        }
+        self.receivable
+            .start_scan(wallet, timestamp, response_skeleton_opt, logger)
+    }
+
+    pub fn finish_payable_scan(&mut self, msg: SentPayables, logger: &Logger) -> PayableScanResult {
+        let scan_result = self.payable.finish_scan(msg, logger);
+        match scan_result.result {
+            OperationOutcome::NewPendingPayable => self.aware_of_unresolved_pending_payable = true,
+            OperationOutcome::Failure => (),
+        };
+        scan_result
+    }
+
+    pub fn finish_pending_payable_scan(
+        &mut self,
+        msg: ReportTransactionReceipts,
+        logger: &Logger,
+    ) -> PendingPayableScanResult {
+        self.pending_payable.finish_scan(msg, logger)
+    }
+
+    pub fn finish_receivable_scan(
+        &mut self,
+        msg: ReceivedPayments,
+        logger: &Logger,
+    ) -> Option<NodeToUiMessage> {
+        self.receivable.finish_scan(msg, logger)
+    }
+
+    pub fn acknowledge_scan_error(&mut self, error: &ScanError, logger: &Logger) {
+        match error.scan_type {
+            ScanType::Payables => {
+                self.payable.mark_as_ended(logger);
+            }
+            ScanType::PendingPayables => {
+                self.pending_payable.mark_as_ended(logger);
+            }
+            ScanType::Receivables => {
+                self.receivable.mark_as_ended(logger);
+            }
+        };
+    }
+
+    pub fn try_skipping_payable_adjustment(
+        &self,
+        msg: BlockchainAgentWithContextMessage,
+        logger: &Logger,
+    ) -> Result<Either<OutboundPaymentsInstructions, PreparedAdjustment>, String> {
+        self.payable.try_skipping_payment_adjustment(msg, logger)
+    }
+
+    pub fn perform_payable_adjustment(
+        &self,
+        setup: PreparedAdjustment,
+        logger: &Logger,
+    ) -> OutboundPaymentsInstructions {
+        self.payable.perform_payment_adjustment(setup, logger)
+    }
+
+    pub fn initial_pending_payable_scan(&self) -> bool {
+        self.initial_pending_payable_scan
+    }
+
+    pub fn unset_initial_pending_payable_scan(&mut self) {
+        self.initial_pending_payable_scan = false
+    }
```

Comment:
Can you please change the name of fn `initial_pending_payable_scan()` to `get_initial_pending_payable_scan()` and `unset_initial_pending_payable_scan(&mut self)` to `set_initial_pending_payable_scan(&mut self, flag: bool)`.